src/webextension/l10n.js

```diff
@@ -23,24 +23,33 @@ async function fetchMessages(baseDir, locale, bundle) {
     const response = await fetch(`${baseDir}/${locale}/${bundle}.ftl`);
     const messages = await response.text();
 
-    return { [locale]: messages };
+    return messages;
   } catch (e) {
-    throw new Error(`unable to fetch localization for ${locale}`);
+    // We couldn't fetch any strings; just return nothing and fluent will fall
+    // back to the default locale if needed.
+    return "";
   }
 }
 
-async function createMessagesGenerator(baseDir, currentLocales, bundle) {
-  const fetched = await Promise.all(
-    currentLocales.map((x) => fetchMessages(baseDir, x, bundle))
-  );
-  const bundles = fetched.reduce(
+async function fetchAllMessages(baseDir, locale, bundles) {
+  return Promise.all(bundles.map((i) => fetchMessages(baseDir, locale, i)));
```

Comment:
Does this need an `await`?

Reply:
No, but the `async` is probably superfluous now.